sale_order_line_discount_validation/report/sale_report.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models
+from odoo.exceptions import Warning as Warn
+
+
+class SaleOrderReport(models.AbstractModel):
+    _name = "report.sale.report_saleorder"
+    _description = "Sale Order Report"
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        docs = self.env["sale.order"].browse(docids)
+        for rec in docs:
+            if rec.order_line.filtered(
+                lambda r: r.discount >= rec.env.company.sale_discount_limit
+            ) and rec.state not in ("approved", "done"):
+                rec._request_approval()
+                raise Warn(
```

Comment:
this change was made due [W0622(redefined-builtin), ] Redefining built-in 'Warning' pylint warning

Reply:
Same as above

Reply:
done